share/functions/__fish_print_VBox_vms.fish

```diff
@@ -0,0 +1,28 @@
+function __fish_print_VBox_vms
+    set -l print_names true
+    set -l print_uuids true
+
+    if set -q argv[1]
+        if test $argv[1] = 'nouuids'
+            set print_uuids false
+        end
+        if test $argv[1] = 'nonames'
+            set print_names false
+        end
+    end
+
+    set -l lines (VBoxManage list vms | string match -r '"(.*)" {(.*)}')
+    while set -q lines[2]
+        if test $print_names = true
+            printf '%s\tVirtual machine\n' $lines[2]
+        end
+
+        if test $print_uuids = true
+            printf '%s\t%s virtual machine\n' $lines[3] $lines[2]
```

Comment:
For the capitalized version, `VBoxManage` is present on Ubuntu, Arch Linux and macOS for the ones I checked. I didn't check on another OS, but I don't see really see a reason why that wouldn't be the case since the [VBoxManage documentation](https://www.virtualbox.org/manual/ch08.html) shows the capitalized version.